lib/ansible/modules/cloud/smartos/vmadm.py

```diff
@@ -0,0 +1,609 @@
+            operate on a VM that is currently provisioned. C(present) means that the VM will be
+            created if it was absent, and that it will be in a running state. C(absent) will
+            shutdown the zone before removing it.
+            C(stopped) means the zone will be created if it doesn't exist already, before shutting
+            it down.
+    autoboot:
+        required: false
+        description:
+          - Whether or not a VM is booted when the system is rebooted.
+    brand:
+        required: true
+        choices: [ joyent, joyent-minimal, kvm, lx ]
+        default: joyent
+        description:
+          - Type for of virtual machine.
+    cpu_cap:
+        required: false
+        description:
+          - Sets a limit on the amount of CPU time that can be used by a VM.
+            Use C(0) for no cap.
+    cpu_shares:
+        required: false
+        description:
+          - Sets a limit on the number of fair share scheduler (FSS) CPU shares for
+            a VM. This limit is relative to all other VMs on the system.
+    customer_metadata:
+        required: false
+        description:
+          - Metadata to be set and associated with this VM, this contain customer
+            modifiable keys.
+    delegate_dataset:
+        required: false
+        description:
+          - Whether to delegate a ZFS dataset to an OS VM.
+    disk_driver:
+        required: false
+        description:
+          - Default value for a virtual disk model for KVM guests.
+    disks:
+        required: false
+        description:
+          - A list of disks to add, valid properties are documented in vmadm(1M).
+    dns_domain:
+        required: false
+        description:
+          - Domain value for C(/etc/hosts).
+    filesystems:
+        required: false
+        description:
+          - Mount additional filesystems into an OS VM.
+    firewall_enabled:
+        required: false
+        description:
+          - Enables the firewall, allowing fwadm(1M) rules to be applied.
+    fs_allowed:
+        required: false
+        description:
+          - Comma separated list of filesystem types this zone is allowed to mount.
+    hostname:
+        required: false
+        description:
+          - Zone/VM hostname.
+    image_uuid:
+        required: false
+        description:
+          - Image UUID.
+    indestructible_zoneroot:
+        required: false
+        description:
+          - Adds an C(@indestructible) snapshot to zoneroot.
+    internal_metadata:
+        required: false
+        description:
+          - Metadata to be set and associated with this VM, this contains operator
+            generated keys.
+    kernel_version:
+        required: false
+        description:
+          - Kernel version to emulate for LX VMs.
+    maintain_resolvers:
+        required: false
+        description:
+          - Resolvers in C(/etc/resolv.conf) will be updated when updating
+            the C(resolvers) property.
```

Comment:
You are definitely confusing markup for options and the values. The `resolvers` is an option so it should be wrapped in `I(..)` in the documentation. The `/etc/resolv.conf` is a value so it should be wrapped in `C(..)` in the documentation. Please go through all the documentation block and fix it accordingly.